### PR TITLE
Implement thread::current()

### PR DESCRIPTION
**Pull request by jamesbornholt**
---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/basic/thread.rs [hunk 1]
```diff
@@ -82,3 +82,36 @@ fn thread_builder_name() {
         100,
     );
 }
+
+#[test]
+fn thread_identity() {
+    check_random(
```
Comment (jorajeev) [inline tests/basic/thread.rs:88]: Why not just `check_dfs` here?

**Review by jorajeev — COMMENTED** [review]
Added a couple of minor comments

File: tests/basic/thread.rs (hunk 1 above)
Comment (jamesbornholt) [reply tests/basic/thread.rs]: Good point, done.

File: src/thread.rs [hunk 2]
```diff
@@ -7,7 +7,7 @@ use crate::runtime::thread::future::ThreadFuture;
 use std::time::Duration;
 
 /// A unique identifier for a running thread
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
```
[the remark this answers is not in the document]
Comment (jamesbornholt) [reply src/thread.rs:10]: To match std's version of [ThreadId](https://doc.rust-lang.org/std/thread/struct.ThreadId.html).

**Review by jorajeev — APPROVED** [review]

**Pull request merged**